### ISSUE TITLE
Speed up the searching/filtering

### ISSUE THREAD

**Issue by kajdehoop** [xrnl/aginun#91]
## Problem that this feature solves
The searching seems to have a delay. Is the server slow?

## Feature description
 Maybe we can do something with caching to speed it up.

**Comment by SamLubbers** [comment]
@kajdehoop good that you mentioned this. The server isn't slow, the issue is that `vue-infinite` doesn't detect that it needs to fetch new roles if the transition hasn't finished, which lead to issue #78. I solved that with the changes in #86, by including a `setTimeout` which waits until the transition has finished before it fetches new roles.

I think that a way to solve this is by doing an [async update](https://vuejs.org/v2/guide/reactivity.html#Async-Update-Queue). So instead of having this:

```js
triggerReload(state) {
  state.infiniteScrollId = !state.infiniteScrollId;
},
```

we would have something like this:

```
triggerReload(state, vm) {
  vm.nextTick(function () {
    state.infiniteScrollId = !state.infiniteScrollId;
  })     
},
```
where `vm`is the `this` object of the component. What this code would do is achieve the same results as the `setTimeout` code in #86, but without the delay.

I can imagine that this can be hard to understand, because you need to have some knowledge about how the `vue-infinite` component works, so I can take care of fixing this.

**Comment by kajdehoop** [comment]
Ah alright! Maybe we can make loading even feel super snappy by preloading the next set of roles. So if you would show role 1-8, you could load roles 9-16 already and show them when you scroll. Then you could preload 17-24, etc. This way the loading would seem instant. Though I think it's a bit early to optimize these kinds of things yet

**Comment by SamLubbers** [comment]
Yes optimization isn't really needed. Without the transition roles load in a fraction of a second :rocket:
